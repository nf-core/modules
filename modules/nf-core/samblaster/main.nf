process SAMBLASTER {
    tag "$meta.id"
    label 'process_low'

    conda "bioconda::samblaster=0.1.26 bioconda::samtools=1.16.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-19fa9f1a5c3966b63a24166365e81da35738c5ab:cee56b506ceb753d4bbef7e05b81e1bfc25d937f-0' :
        'biocontainers/mulled-v2-19fa9f1a5c3966b63a24166365e81da35738c5ab:cee56b506ceb753d4bbef7e05b81e1bfc25d937f-0' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.bam"), emit: bam
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
    def args3 = task.ext.args3 ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    if( "$bam" == "${prefix}.bam" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    samtools view -h $args2 $bam | \\
    samblaster $args | \\
    samtools view $args3 -Sb - >${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samblaster: \$( samblaster -h 2>&1 | head -n 1 | sed 's/^samblaster: Version //' )
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """
}
