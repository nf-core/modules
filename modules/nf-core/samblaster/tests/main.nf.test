nextflow_process {

    name "Test Process SAMBLASTER"
    script "../main.nf"
    process "SAMBLASTER"

    tag "modules"
    tag "modules_nfcore"
    tag "samblaster"

    test("homo_sapiens-test_paired_end_umi_unsorted_bam") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_umi_unsorted_bam'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_umi_unsorted_bam'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
