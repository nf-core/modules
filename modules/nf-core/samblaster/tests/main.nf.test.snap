{
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a70467f2dfc2e0d8e81787223d2fc77"
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a70467f2dfc2e0d8e81787223d2fc77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-26T14:04:42.510824"
    },
    "homo_sapiens-test_paired_end_umi_unsorted_bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,634a6bd541478e970f0a4c279f399889"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a70467f2dfc2e0d8e81787223d2fc77"
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,634a6bd541478e970f0a4c279f399889"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a70467f2dfc2e0d8e81787223d2fc77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-26T14:04:38.118875"
    }
}